source/rhoas/src/main/java/com/openshift/cloud/RHOASOperator.java

```diff
@@ -69,9 +71,43 @@ public int run(String... args) throws Exception {
     config = configuration.getConfigurationFor(serviceRegistryConnectionController);
     LOG.info("CR class: " + config.getCustomResourceClass());
 
+    cleanupDeletedLegacyFinalizers();
+
     operator.start();
 
     Quarkus.waitForExit();
     return 0;
   }
+
+  private void cleanupDeletedLegacyFinalizers() {
```

Comment:
We can keep it for couple releases and remove it.
Maybe we can have issue to remove it